src/module_wrap.cc

```diff
@@ -622,6 +628,50 @@ void ModuleWrap::SetImportModuleDynamicallyCallback(
   iso->SetHostImportModuleDynamicallyCallback(ImportModuleDynamically);
 }
 
+void ModuleWrap::InitializeImportMetaObject(Local<Context> context,
+                                            Local<Object> meta) {
+  Isolate* isolate = context->GetIsolate();
+  Environment* env = Environment::GetCurrent(isolate);
+  meta->CreateDataProperty(context, env->url_string(),
```

Comment:
i'm wondering if there's any perf hit with moving this to the js callback. if there isn't i would suggest doing that just for consistency.

Reply:
You mean by passing the URL string to the JS callback?

Reply:
no i mean like
```js
importMetaCallback(wrap, meta) {
  // attach wrap.url to meta in a safe way here
}
```